webapp/tests/forms/steps/test_steuerlotse_step.py

```diff
@@ -417,14 +485,15 @@ def setUp(self) -> None:
 
     def test_if_post_then_return_redirect_to_next_step(self):
         next_step = MockFinalStep
+        should_update_data = True # Set flags to indicate, that this is a post request
         with self.app.test_request_context(
                 path="/" + self.endpoint_correct + "/step/" + MockFormStep.name) as req:
             req.session = SecureCookieSession({'form_data': create_session_form_data(self.session_data)})
-            form_step = MockFormStep(endpoint=self.endpoint_correct, stored_data={}, next_step=next_step)
-
-            # Set flags to indicate, that this is a post request with valid data
+            form_step = MockFormStep(endpoint=self.endpoint_correct, stored_data={}, next_step=next_step,
+                                     render_info=MockFormStep.prepare_render_info({},
+                                                                                  input_data=ImmutableMultiDict({}),
+                                                                                  should_update_data=should_update_data))
             form_step.should_update_data = True
-            form_step.data_is_valid = True
 
             response = form_step.handle()
```

Comment:
I don't get it I think. Why do you set shoul_update_Data in l.488 but also in l.496? And - why even setting is as a variable at all and not just in the `prepare_render_info` call?

Reply:
Yep, that helped. I just meant picking the value of the attribute from the render_info in the `init`. But we will  then keep it mostly that way